File: lab/python/arrays_e_listas/busca_linear.py
def busca_linear1(lista, valor):
    for i, item in enumerate(lista):
        if valor == item:
            return i
    return -1


def busca_linear2(lista, valor):
    i = 0
    for item in lista:
        if valor == item:
            return i
        i += 1
    return -1


lista = [2, 4, 6, 8, 10]
valor = 6

print("Posição:", busca_linear1(lista, valor))
print("Posição:", busca_linear2(lista, valor))

File: lab/javascript/arrays_e_listas/busca_linear.js
function busca_linear1(lista, valor) {
  for (let i = 0; i < lista.length; i++) {
    if (lista[i] === valor) {
      return i;
    }
  }
  return -1;
}

function busca_linear2(lista, valor) {
  let i = 0;
  for (const item of lista) {
    if (item === valor) {
      return i;
    }
    i++;
  }
  return -1;
}

// Teste
const lista = [2, 4, 6, 8, 10];
const valor = 6;

console.log("Posição:", busca_linear1(lista, valor));
console.log("Posição:", busca_linear2(lista, valor));
